### PR TITLE
[wip] Add support for sbt files

### PR DESCRIPTION
**Pull request by jvican**

### CODE REVIEW

File: sbthost/nsc/src/main/scala/scala/meta/internal/sbthost/SbthostPipeline.scala [hunk 1]
```diff
@@ -74,11 +74,14 @@ trait SbthostPipeline extends DatabaseOps { self: SbthostPlugin =>
           case els =>
             Paths.get(els.file.getAbsoluteFile.toURI)
         }
+        // Select Sbt0137 dialect for scala sources extracted from sbt files
+        val detectedDialect =
+          if (g.getClass.getName.contains("sbt.compiler.Eval")) "Sbt0137" else "Scala210"
```
Comment (olafurpg) [inline sbthost/nsc/src/main/scala/scala/meta/internal/sbthost/SbthostPipeline.scala:79]: Nice!

File: sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala [hunk 2]
```diff
@@ -4,64 +4,128 @@ package tests
...
+class SbtFileTest
+    extends SbthostTest(
+      RelativePath("META-INF/semanticdb/build.sbt.semanticdb"),
+      s"""build.sbt
+         |---------
+         |Dialect:
+         |Sbt0137
+         |
+         |Names:
+         |[0..12): organization => _root_._empty_.
```
Comment (olafurpg) [inline sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala:72]: Interesting, when running under sbt each statement becomes a separate compilation unit

```
[info] ====================
[info] = build.sbt
[info] ====================
[info] val p1 = project
[info]
[info]
[info]
[info] ====================
[info] = /private/var/folders/30/6jlz_xfs46ndvn212mt_wj6m0000gn/T/sbt_8a2705d2/basic/build.sbt
[info] ====================
[info] name := "basic"
[info]
[info]
[info] ====================
[info] = /private/var/folders/30/6jlz_xfs46ndvn212mt_wj6m0000gn/T/sbt_8a2705d2/basic/build.sbt
[info] ====================
[info] organization := "me.vican.jorge"
[info]
```

Comment (olafurpg) [reply sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala:72]: We'll have to figure a way merge these into a single .semanticdb.

File: sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala [hunk 3]
```diff
@@ -4,64 +4,128 @@ package tests
...
+         |[13..15): := => _root_.sbt.internals.DslEntry.fromSettingsDef(Lsbt/Init/SettingsDefinition;)Lsbt/internals/DslEntry;.
+         |
+         |Denotations:
```
Comment (olafurpg) [inline sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala]: I propose we leave denotations out of these tests

```scala
    val noDenotations = mirror.database.copy(entries = mirror.entries.map {
      case (input, attrs) =>
        input -> attrs.copy(denotations = Nil)
    })
    val syntax = noDenotations.syntax
    assert(syntax.trim == expectedSyntax.trim)
```

That avoid the need for the StableHashId and pre-process tricks, which was causing flakiness when I ran the tests multiple times locally. We are mostly interested in the positions in the Names section, anyways. WDYT?

**Review by olafurpg — COMMENTED** [review]
This is really nice, I can run `~test`, edit sbthost/nsc and "immediately" see the semanticdb from build.sbt. Awesome work with the sbt foo.

File: sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala (hunk 2 above)
Comment (olafurpg) [reply sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala:72]: Let's leave that for another PR.

Comment (jvican) [reply sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala:72]: @olafurpg Yes, sbt creates a compilation unit for every unit. I have a WIP PR from two months ago that put everything into only one compilation unit, but it's not ready yet, so I think working around this issue in sbthost is the way to go :smile

File: sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala (hunk 3 above)
Comment (jvican) [reply sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala]: I thought about this too. I'm fine with this, I hope there are no future regressions!

Comment (olafurpg) [reply sbthost/tests/src/test/scala/scala/meta/tests/SbthostTest.scala]: We can test the denotations separately for .scala files. For .sbt files the most important bit is the positions of names.

**Comment by olafurpg** [comment]
Superseded by #8

**Pull request closed without merging**